### PR TITLE
Changed sf::Event::Type to sf::Event::EventType for easier compatibli…

### PR DESCRIPTION
**Pull request by JonnyPtn**
Allows projects to switch back to SFML without errors (assuming c++11 capable compiler)

This is toggleable by enabling the MML_SFML_COMPAT cmake option

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,6 +91,9 @@ if(SFML_BUILD_WINDOW)
     sfml_set_option(SFML_OPENGL_ES ${OPENGL_ES} BOOL "TRUE to use an OpenGL ES implementation, FALSE to use a desktop OpenGL implementation")
 endif()
 
+# add an option for SFML compatibility, temporarily
+sfml_set_option(MML_SFML_COMPAT FALSE BOOL "TRUE to include API alterations allowing compatibility with SFML")
```
Comment (eliasdaler) [inline CMakeLists.txt:95]: docstring should better be "Include API alterations allowing compability with SFML"

People will get that "TRUE" means "ON" :)

Comment (Ceylo) [reply CMakeLists.txt:95]: Maybe but… this would be inconsistent with all other docstrings

File: include/SFML/Window/Mouse.hpp [hunk 2]
```diff
@@ -58,6 +59,8 @@ class SFML_WINDOW_API Mouse
 
         Count       ///< Keep last -- the total number of mouse buttons
     };
+    
+    MML_COMPAT_ALIAS(ButtonCount, Button::Count)
```
Comment (eliasdaler) [inline include/SFML/Window/Mouse.hpp]: Let's bring back ButtonCount into Button enum, maybe? That way, we can remove MML_COMPAT_ALIAS for now and make things simpler. We're unlikely to need another "aliases" like this.

File: include/SFML/Window/Keyboard.hpp [hunk 3]
```diff
@@ -161,6 +162,8 @@ class SFML_WINDOW_API Keyboard
         SemiColon = Semicolon,    ///< \deprecated Use Semicolon instead
         Return    = Enter         ///< \deprecated Use Enter instead
     };
+    
+    MML_COMPAT_ALIAS(KeyCount, Key::Count)
```
Comment (eliasdaler) [inline include/SFML/Window/Keyboard.hpp:166]: Same here as for Mouse.hpp

File: src/SFML/Window/CMakeLists.txt [hunk 4]
```diff
@@ -242,6 +242,12 @@ if ((NOT ${CMAKE_VERSION} VERSION_LESS 3.11) AND (NOT OpenGL_GL_PREFERENCE))
     set(OpenGL_GL_PREFERENCE "LEGACY")
 endif()
 
+# Compatibility for:
+# - sf::Event::EventType
+if(MML_SFML_COMPAT)
```
Comment (eliasdaler) [inline src/SFML/Window/CMakeLists.txt:247]: Let's move it to sfml_add_library macro. We'll already use MML_SFML_COMPAT in sfml-graphics too, so it's better to add this definition to all SFML targets.

File: CMakeLists.txt (hunk 1 above)
Comment (eliasdaler) [reply CMakeLists.txt]: Ok :)

**Review by eliasdaler — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (eliasdaler) [reply CMakeLists.txt:95]: Ok :)

**Pull request merged**